### PR TITLE
[AzureRT] Sprint 56 update

### PR DESCRIPTION
**Pull request by hyonholee**
1. Enable BGInfo by default
2. Fix the issue when an OS disk is in a different resource group - Now New-AzureRmVM does not create a new storage account for boot diagnostics.
3. Add a new test when an empty list for additional unattend content is set for Update-AzureRmVM.
4. Re-record all compute tests because of New-AzureRmVM behavior change.

### CODE REVIEW

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/332/

File: src/ResourceManager/Compute/Commands.Compute/Common/ComputeClientBaseCmdlet.cs [hunk 1]
```diff
@@ -63,6 +63,7 @@ protected void ExecuteClientAction(Action action)
             }
             catch (CloudException ex)
             {
+                base.EndProcessing();
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Common/ComputeClientBaseCmdlet.cs]: What if ```EndProcessing()``` shows exception? In that case, ```ComputeCloudException``` will be missed.

**Comment by huangpf** [comment]
LGTM, other than 1 question about ```EndProcessing```.
